digits/model/tasks/caffe_train.py

```diff
@@ -860,7 +860,10 @@ def _pycaffe_args(self, gpu_id):
             solver_type = solver_type_mapping[self.solver_type]
         except KeyError:
             raise ValueError("Unknown solver type {}.".format(self.solver_type))
-        gpu_script = "caffe.set_device({id});".format(id=gpu_id) if gpu_id else ""
+        if gpu_id:
```

Comment:
Seems that gpu_id is a string, not an int.  But you are right, testing it against None is a better way to understand the program logic.